jobmon_client/src/jobmon/client/task_resources.py

```diff
@@ -118,11 +124,30 @@ def adjust_resources(
         resource_updates: Dict[str, Any] = {}
 
         # Only cores, memory, and runtime get scaled
-        for resource, scaling_factor in resource_scales.items():
+        for resource, scaler in resource_scales.items():
             if resource in existing_resources.keys():
-                resource_updates[resource] = self.scale_val(
-                    existing_resources[resource], scaling_factor
-                )
+                if isinstance(scaler, numbers.Number):
+                    new_resource_value = self.scale_val(
+                        existing_resources[resource], scaler
+                    )
+                elif callable(scaler):
+                    new_resource_value = scaler(existing_resources[resource])
+                elif isinstance(scaler, Iterator):
+                    new_resource_value = next(scaler)
```

Comment:
You might want to handle `StopIteration` here, what's the expectation if an iterator is too short to handle the amount of retries?

Reply:
I'll add a warning and re-use the old value